python/pyspark/sql/readwriter.py

```diff
@@ -520,20 +537,24 @@ def func(iterator):
             raise TypeError("path can be only string, list or RDD")
 
     @since(1.5)
-    def orc(self, path, mergeSchema=None, recursiveFileLookup=None):
+    def orc(self, path, mergeSchema=None, pathGlobFilter=None, recursiveFileLookup=None):
         """Loads ORC files, returning the result as a :class:`DataFrame`.
 
         :param mergeSchema: sets whether we should merge schemas collected from all
             ORC part-files. This will override ``spark.sql.orc.mergeSchema``.
             The default value is specified in ``spark.sql.orc.mergeSchema``.
+        :param pathGlobFilter: an optional glob pattern to only include files with paths matching
+                               the pattern. The syntax follows `org.apache.hadoop.fs.GlobFilter`.
+                               It does not change the behavior of `partition discovery`_.
         :param recursiveFileLookup: recursively scan a directory for files. Using this option
-            disables `partition discovery`_.
+                                    disables `partition discovery`_.
```

Comment:
Style nit: This does make the indentation more consistent with the rest of the file, but I wish we could start moving the indentation style to fit the pre-change indentation.

Right now these doc strings waste a lot of leading whitespace and have wildly varying indentation that depends on the length of the parameter name. It would be better if we just indented continuing lines with a consistent 4 spaces, like how `mergeSchema`'s docstring looks.

Reply:
Yeah, valid point. Let fix them consistent for now, and maybe we can handle them in a batch separately.

Reply:
Would you be open to a batch update of all docstrings to tweak the indentation? Or would it be considered as adding too much noise to the git history? I assumed we didn't want to make sweeping changes like that, so my idea was to update the indentation bit by bit as part of more significant changes, which is what I did with `recursiveFileLookup` and `mergeSchema` when I added those in.

Reply:
Since we're going ahead for Spark 3, we wont likely backport many things that cause conflicts. So I was thinking it's a feasible option.

Also, I think we might have to document this first that virtical alignment isn't preferred. I think virtical alignment is still valid per PEP 8 and PEP 257.

One downside of doing bit by bit is a confusion by mixed style. Considering that we wont likely add many new docstrings, mixed style exists in a long term.

Reply:
So .. if you dont mind, I would like this run that separately :-)..